src/node_report.cc

```diff
@@ -0,0 +1,1115 @@
+
+  for (size_t i = 0; i < arraysize(rlimit_strings); i++) {
+    if (getrlimit(rlimit_strings[i].id, &limit) == 0) {
+      out << "  " << rlimit_strings[i].description << " ";
+      if (limit.rlim_cur == RLIM_INFINITY) {
+        out << "       unlimited";
+      } else {
+#if defined(_AIX) || defined(__sun)
+        snprintf(buf, sizeof(buf), "%16ld", limit.rlim_cur);
+        out << buf;
+#elif(defined(__linux__) && !defined(__GLIBC__))
+        snprintf(buf, sizeof(buf), "%16lld", limit.rlim_cur);
+        out << buf;
+#else
+        snprintf(buf, sizeof(buf), "%16" PRIu64, limit.rlim_cur);
+        out << buf;
+#endif
+      }
+      if (limit.rlim_max == RLIM_INFINITY) {
+        out << "       unlimited\n";
+      } else {
+#if defined(_AIX)
+        snprintf(buf, sizeof(buf), "%16ld\n", limit.rlim_max);
+        out << buf;
+#elif(defined(__linux__) && !defined(__GLIBC__))
+        snprintf(buf, sizeof(buf), "%16lld\n", limit.rlim_max);
+        out << buf;
+#else
+        snprintf(buf, sizeof(buf), "%16" PRIu64 "\n", limit.rlim_max);
+        out << buf;
+#endif
+      }
+    }
+  }
+#endif
+
+  out << "\nLoaded libraries\n";
+  PrintLoadedLibraries(out, isolate);
+}
+
+/*******************************************************************************
+ * Functions to print a list of loaded native libraries.
+ *
+ ******************************************************************************/
+#ifdef __linux__
+static int LibraryPrintCallback(struct dl_phdr_info* info,
+                                size_t size, void* data) {
+  std::ostream* out = reinterpret_cast<std::ostream*>(data);
+  if (info->dlpi_name != nullptr && *info->dlpi_name != '\0') {
+    *out << "  " << info->dlpi_name << "\n";
+  }
+  return 0;
+}
+#endif
+
+static void PrintLoadedLibraries(std::ostream& out, Isolate* isolate) {
+#ifdef __linux__
+  dl_iterate_phdr(LibraryPrintCallback, &out);
+#elif __APPLE__
+  int i = 0;
+  const char* name = _dyld_get_image_name(i);
+  while (name != nullptr) {
+    out << "  " << name << "\n";
+    i++;
+    name = _dyld_get_image_name(i);
+  }
+#elif _AIX
+  // We can't tell in advance how large the buffer needs to be.
+  // Retry until we reach too large a size (1Mb).
+  const unsigned int buffer_inc = 4096;
+  unsigned int buffer_size = buffer_inc;
+  char* buffer = reinterpret_cast<char*>(malloc(buffer_size));
+  int rc = -1;
+  while (buffer != nullptr && rc != 0 && buffer_size < 1024 * 1024) {
+    rc = loadquery(L_GETINFO, buffer, buffer_size);
+    if (rc == 0) {
+      break;
+    }
+    free(buffer);
+    buffer_size += buffer_inc;
+    buffer = reinterpret_cast<char*>(malloc(buffer_size));
+  }
+  if (buffer == nullptr) {
+    return;  // Don't try to free the buffer.
+  }
+  if (rc == 0) {
+    char* buf = buffer;
+    ld_info* cur_info = nullptr;
+    do {
+      cur_info = reinterpret_cast<ld_info*>(buf);
+      char* member_name = cur_info->ldinfo_filename
+        + strlen(cur_info->ldinfo_filename) + 1;
```

Comment:
style nit: 4 spaces indentation, operand goes ion the previous line

Reply:
done